src/compiler/commandLineParser.ts

```diff
@@ -2477,6 +2477,13 @@ namespace ts {
             parseOwnConfigOfJson(json, host, basePath, configFileName, errors) :
             parseOwnConfigOfJsonSourceFile(sourceFile!, host, basePath, configFileName, errors);
 
+        if (ownConfig.options?.paths) {

```

Comment:
I dont think this is going to work when paths is in extended config file since the paths should be resolved with baseUrl = directory of extended config file. So you would need to handle that magic somewhere for extended config path

Reply:
It does work; see [the test](https://github.com/microsoft/TypeScript/pull/40101/files#diff-0917ba20b8847b85fdb7085454a168da) for it

Reply:
(It works because this function recurses on extended configs, and this line happens on the way up out of that recursion.)